src/merkle/tiered_smt/mod.rs

```diff
@@ -0,0 +1,211 @@
+use super::{BTreeMap, Felt, Rpo256, RpoDigest, Serializable, Vec, Word};
+
+mod bits;
+pub use bits::BitsIterator;
+
+mod content;
+pub use content::{Content, ContentType};
+
+mod index;
+pub use index::TreeIndex;
+
+mod storage;
+pub use storage::{Storage, StorageError};
+
+#[cfg(test)]
+mod tests;
+
+// TIERED SPARSE MERKLE TREE
+// ================================================================================================
+
+pub struct TieredSmt {
+    // TODO this should be a constant
+    empty_subtrees: Vec<RpoDigest>,
+    storage: Storage,
+}
+
+impl TieredSmt {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    pub const MAX_DEPTH: usize = 64;
+    pub const TIER_DEPTH: usize = 4;
+
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    pub fn with_storage(storage: Storage) -> Self {
+        // Construct empty node digests for each layer of the tree
+        let empty_subtrees = (0..Self::MAX_DEPTH + 1)
+            .scan(Word::default().into(), |state, _| {
```

Comment:
I guess that `Word` and `RpoDigest` are the same thing, but shouldn't this use the default of `RpoDigest` instead? Since that is the default of a `Content` with `ContentType::Empty`?

Reply:
That makes sense, it's a leftover of the previous implementation. Fixed!